### DISCUSSION TITLE
Email Markup Goals

### DISCUSSION THREAD

**Discussion by M-J-Robbins** [email-markup-consortium/email-markup-consortium#27] [Ideas]
In the meeting on 12th October, we discussed splitting into groups to focus on a few projects. This discussion is for the top level Email Markup Goals project.

This project is to define what we mean when we talk about improving email markup, what standards should that markup strive for. 

This guide is top level and should not go into details of what markup to use.  It's about what we want to achieve, rather than how we are going to achieve it. Although we may need to use examples to help explain the ideas, that should be in the notes and not the guide.

A second document will come after this *Email Markup Guidelines for Developers* that will explain how to get as close to these standards as possible with the current available supported markup. This second document will be very similar to Rémi's [Email Guidelines](https://github.com/hteumeuleu/email-guidelines).

I've made a few notes here to get us started.
https://docs.google.com/document/d/1QsiUt8ryUt1-rXg2YbKyKGB3JyI7ipBQUTY0M8kT-UQ/edit#

## Get involved
* Read of the [Google Doc](https://docs.google.com/document/d/1QsiUt8ryUt1-rXg2YbKyKGB3JyI7ipBQUTY0M8kT-UQ/edit#)
* Add some notes to the doc
* Add some thoughts in this discussion
* Let me know if you are interested in being a key member of this project and I'll add your name to the list below
  * Project members are the key contributors to this project, but all members of the EMC can still add comments you don't need to be a project member.

## Project members
* @M-J-Robbins 
* @alicemeichi 
* @dylanatsmith 
* @husseinalhammad
* @mattdionne 
* @megbosh 
* @sugarfreq

**Comment by Malvoz** [comment]
A comment on (and perhaps addition to) the Reliability section, relating to these points:

> - Code structure should remain intact when unsupported code is removed.
> - Unsupported elements should keep their supported attributes and replaces with a semantically natural element
> (replace `<main class=”foo”>` with `<div class=”foo”>`

Clients that _rename elements_ that they do not support, such as https://www.caniemail.com/features/html-form/:

> 4. Not supported. `<form>` is transformed into `<noform>`.

should probably be recommended to rename elements using [valid custom element names](https://html.spec.whatwg.org/multipage/custom-elements.html#valid-custom-element-name) (i.e. `<no-form>`, or just turn them into `div` elements). Having clients rename `<element>` to `<noElement>` can cause web platform compatibility issues: 1) may disuade WHATWG from introducing an element in HTML with a particular name because it is already used widely in emails by some clients, or 2) if HTML did introduce `<noElement>` then that may break things in these emails.

**Reply by Malvoz** [reply]
> A comment on (and perhaps addition to) the Reliability section

OTOH perhaps that speaks to having a new section:

> ### Web Platform Compatibility
>
> - Clients should not rename/transform elements that they do not support to element names other than `div`, `span` or [valid custom element names](https://html.spec.whatwg.org/multipage/custom-elements.html#valid-custom-element-name)
> - Clients should not create non-vendor prefixed CSS properties or values
> - Clients should not create new HTML attributes

**Reply by M-J-Robbins** [reply]
Thanks @Malvoz some good suggestions, I've added them to the doc and also added a couple of comments.

**Comment by M-J-Robbins** [comment]
I've added a deadline of 2nd November on the doc here https://docs.google.com/spreadsheets/d/1qCKFSdbc7xBGk84YBJ6ZQvnHB0lK8IweRXdXu-W5veU/edit#gid=0 

This deadline is only for the first draft, we can still make updates after that.  
I imagine as other project build and want to link back to this document we'll get new ideas of things to add.

My thoughts for the requirements of the first draft
* It should set up the structure for the document
* It should be set up in a way so offshoot projects can link back to it easily.
* Shorter is better. It's easy to add to it, removing things might get complicated if other things are referencing it.

Let me know any thoughts.

**Reply by Malvoz** [reply]
> removing things might get complicated if other things are referencing it.

This happens all the time in W3C specs, a changelog section may be helpful. Here's an example: https://www.w3.org/TR/payment-request/#changelog

**Reply by M-J-Robbins** [reply]
Oh nice that's very useful, I'll add a changelog title so we're ready for when this moves to Github.

**Comment by M-J-Robbins** [comment]
Planning a call to discuss the document and get it ready for first draft.

Call will be in Tuesday 26th October at [5pm BST](https://www.worldtimebuddy.com/?qm=1&lid=14,5,8&h=14&date=2021-10-26&sln=17-18&hf=1) if anyone else wants to be added to the call let me know. 

After the call I'm hoping we'll be in a good place and still have a week to fine tune things if needed.

Next step will be to publish it as a draft in this github repo and start building additional documents off this one.
